### ISSUE TITLE
regression in quantity parsing in 4.7.0

### ISSUE THREAD

**Issue by onursatici** [fabric8io/kubernetes-client#1953]
Hello,
There is a regression in QuantityBuilder that breaks amounts specified with format suffixes:
pre 4.7.0 parsed `new QuantityBuilder().withAmount("500m").build()` into a Quantity with `500` as amount and `m` as format, but now we get `null` as format, and this is serialised as `500` instead of `500m`

The generated `QuantityBuilder` has:
```
    public Quantity build(){
            Quantity buildable = new Quantity(fluent.getAmount());
            buildable.setFormat(fluent.getFormat());
            return buildable;
    }
```
which overrides the format of `buildable` with `fluent.getFormat()` which is null.

An example use in spark which will be affected by this regression:
https://github.com/apache/spark/blob/master/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala#L91-L96

**Comment by rohanKanojia** [comment]
I recently fixed #885 . But I feel that old behaviour was confusing. You're setting the amount only. Does it work when you do `new Quantity("500m")`?

**Comment by onursatici** [comment]
Yes it works when using the constructor. I agree it was confusing before, but it is a backcompat break, and a subtle one, as the builder doesn't throw, but silently drop the format.
